src/meltano/cli/config.py

```diff
@@ -372,14 +375,29 @@ def set_(
     value: t.Any,
     store: str,
     interactive: bool,
+    from_file,
 ):
     """Set the configurations' setting `<name>` to `<value>`."""
     if len(setting_name) == 1:
         setting_name = tuple(setting_name[0].split("."))
 
     interaction = InteractiveConfig(ctx=ctx, store=store, extras=False)
+
     if interactive:
-        interaction.configure_all()
+        if sys.stdin.isatty():
+            # Interactive stdin flow
+            interaction.configure_all()
```

Comment:
Interaction can only be explicitly requested so checking `sys.stdin.isatty()` feels redundant.

```suggestion
        interaction.configure_all()
        ctx.exit()
```